.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+
+# Runs build and test on:
+#   every push that has a change in a file not in the docs folder
+#   every pull request with main branch as the base that has a change in a file not in the docs folder
+on:
+  push:
+    paths:
+      - '**'
+      - '!docs/**'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**'
+      - '!docs/**'
+
+jobs:
+  build:
+    name: Check and build codebase
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 15.x]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Get Yarn Cache Directory Path
+        id: cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Use Yarn Cache
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: ${{ steps.cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ matrix.node-version }}-${{ hashFiles('**/yarn.lock') }}
```

Comment:
Minor - will `**/yarn.lock` also include and hash `docs/yarn.lock`? Because we wouldn't want changes in `docs/yarn.lock` to invalidate the cache for this workflow which only deals with the root install.

Reply:
Good eye.  I updated it to be specific.